python_modules/dagster/dagster/core/definitions/event_metadata/table.py

```diff
@@ -0,0 +1,251 @@
+        ],
+    )
+):
+    """Descriptor for "table-level" constraints. Presently only one property,
+    `other` is supported. This contains strings describing arbitrary
+    table-level constraints. A table-level constraint is a constraint defined
+    in terms of multiple columns (e.g. col_A > col_B) or in terms of rows.
+
+    Args:
+        other (List[str]): Descriptions of arbitrary table-level constraints.
+    """
+
+    def __new__(
+        cls,
+        other: List[str],
+    ):
+        return super(TableConstraints, cls).__new__(
+            cls,
+            other=check.list_param(other, "other", of_type=str),
+        )
+
+
+_DefaultTableConstraints = TableConstraints(other=[])
+
+# ########################
+# ##### TABLE COLUMN
+# ########################
+
+
+@whitelist_for_serdes
+class TableColumn(
+    NamedTuple(
+        "TableColumn",
+        [
+            ("name", str),
+            ("type", str),
+            ("description", Optional[str]),
+            ("constraints", "TableColumnConstraints"),
+        ],
+    )
+):
+    """Descriptor for a table column. The only property that must be specified
+    by the user is `name`. If no `type` is specified, `string` is assumed. If
+    no `constraints` are specified, the column is assumed to be nullable
+    (i.e. `required = False`) and have no other constraints beyond the data type.
+
+    Args:
+        name (List[str]): Descriptions of arbitrary table-level constraints.
+        type (Optional[str]): The type of the column. Can be an arbitrary
+            string. Defaults to `"string"`.
+        description (Optional[str]): Description of this column. Defaults to `None`.
+        constraints (Optional[TableColumnConstraints]): Column-level constraints.
+            If unspecified, column is nullable with no constraints.
+    """
+
+    def __new__(
+        cls,
+        name: str,
+        type: str = "string",  # pylint: disable=redefined-builtin
+        description: Optional[str] = None,
+        constraints: Optional["TableColumnConstraints"] = None,
+    ):
+        return super(TableColumn, cls).__new__(
+            cls,
+            name=check.str_param(name, "name"),
+            type=check.str_param(type, "type"),
+            description=check.opt_str_param(description, "description"),
+            constraints=cast(
+                "TableColumnConstraints",
+                check.opt_inst_param(
+                    constraints,
+                    "constraints",
+                    TableColumnConstraints,
+                    default=_DefaultTableColumnConstraints,
+                ),
+            ),
+        )
+
+
+# ########################
+# ##### TABLE FIELD CONSTRAINTS
+# ########################
+
+
+@whitelist_for_serdes
+class TableColumnConstraints(
+    NamedTuple(
+        "TableColumnConstraints",
+        [
+            ("nullable", bool),
+            ("unique", bool),
+            ("other", Optional[List[str]]),
```

Comment:
Prefer Sequence to List

Reply:
Do you mean for this specific item (`other` in the `NamedTuple`), or for the various places I've used `List` in the new metadata types. Also, why is `Sequence` preferred? Isn't it essentially a union of `List` and a few other collection types?

I haven't seen `Sequence` much used in the parts of the code I'm familiar with, but I just grepped and see that it's used in `SolidDefinition` and other places. But in `SolidDefinition` it actually looks incorrect, because there's a runtime `check.list_param` call in the constructor (just as there is in the `TableMetadata` code), which will throw an error for a tuple (which is compatible with `Sequence`).